tests/test_client.py

```diff
@@ -402,6 +403,67 @@ def test_request_invalid_http_headers():
     assert str(value_error.value).startswith("cannot override reserved HTTP header")
 
 
+def test_enabling_https_automatically_when_using_port_443(monkeypatch):
+    post_recorder = ArgumentsRecorder()
+    monkeypatch.setattr(TrinoRequest.http.Session, "post", post_recorder)
+
+    req = TrinoRequest(
+        host="coordinator",
+        port=constants.DEFAULT_TLS_PORT,
+        user="test",
+        catalog="test",
+        schema="test",
+        session_properties={},
```

Comment:
Let's remove unnecessary arguments and use literal because actual users won't use the constant. 
```python
    req = TrinoRequest(
        host="coordinator",
        port=443,
        user="test",
    )
```